oauth2_provider/oauth2_validators.py

```diff
@@ -300,7 +300,10 @@ def validate_client_id(self, client_id, request, *args, **kwargs):
         return self._load_application(client_id, request) is not None
 
     def get_default_redirect_uri(self, client_id, request, *args, **kwargs):
-        return request.client.default_redirect_uri
+        try:
+            return request.client.default_redirect_uri
+        except Exception as e:
+            raise HttpResponseBadRequest(e)
```

Comment:
HttpResponseBadRequest is not an Exception. You probably want to raise [BadRequest](https://docs.djangoproject.com/en/3.2/ref/exceptions/#badrequest).

Reply:
And the test should be for a more specific Exception.